File: apzkr-pzpi-21-10-karasov-mykhailo/Task4-MobileClient/app/src/main/java/com/example/tasksyncmobileapp/model/Activity.kt
package com.example.tasksyncmobileapp.model

data class Activity(
    val id: Int,
    val activityTitle: String,
    val description: String,
    val requiredWorkerCount: Int,
    val timeShift: Int,
    val complexityId: Int,
    val educationId: Int,
    val companyId: Int,
    val complexity: Complexity,
    val education: Education,
    val company: Company,
    val users: List<User>
)

File: apzkr-pzpi-21-10-karasov-mykhailo/Task1-Server/src/core/domain/models/Acitivity/Activity.ts
import ComplexityDomainModel from "../Complexity/Complexity";
import EducationDomainModel from "../Education/Education";
import CompanyDomainModel from "../Company/Company";
import UserDomainModel from "../User/User";


export default class ActivityDomainModel {
    constructor(
        public readonly id: number,
        public readonly activityTitle: string,
        public readonly description: string | null,
        public readonly requiredWorkerCount: number,
        public readonly timeShift: number,
        public readonly complexityId: number | null,
        public readonly educationId: number | null,
        public readonly companyId: number | null,
        public readonly complexity: ComplexityDomainModel | null,
        public readonly education: EducationDomainModel | null,
        public readonly company: CompanyDomainModel | null,
        public readonly users: UserDomainModel[]
    ) {}
}